src/luarocks/core/types/build.d.tl

```diff
@@ -0,0 +1,66 @@
+local record build
+
+   interface Build
+      record Install
+         lua: {(string|integer): string}
+         lib: {(string|integer): string}
+         conf: {(string|integer): string}
+         bin: {(string|integer): string}
+      end
+      
+      type: string
+      install: Install
+      copy_directories: {string}
+      patches: {string : string}
+      extra_files: {string : string}
+      macosx_deployment_target: string
+   end
+
+   record BuiltinBuild
+      is Build
+      where self.type == "builtin"
+
+      record Module
+         is {string}
+         sources: string | {string}
+         libraries: string | {string}
+         defines: {string}
+         incdirs: {string}
+         libdirs: {string}
+      end
+
+      modules: {string: (string | Module)}
+   end
+
+
+   record CMakeBuild
```

Comment:
all these records (BuiltinBuild, CMakeBuild, etc.) should be in the luarocks/build module that uses them — let's keep only the abstract interface here